src/Components/Assets/AssetType/HL7Monitor.tsx

```diff
@@ -14,31 +14,103 @@ interface HL7MonitorProps {
   assetId: string;
   asset: any;
 }
+const initForm = {
+  middleware_hostname: {
+    value: "",
+    error: "",
+  },
+  localipAddress: {
+    value: "",
+    error: "",
+  },
+};
+const initialState = {
+  form: { ...initForm },
+};
+const hl7MonitiorFormReducer = (state = initialState, action: any) => {
+  switch (action.type) {
+    case "set_form": {
+      return {
+        ...state,
+        form: action.form,
+      };
+    }
+    default:
+      return state;
+  }
+};
 
 const HL7Monitor = (props: HL7MonitorProps) => {
+  const [state, dispatch_form] = useReducer(
+    hl7MonitiorFormReducer,
+    initialState
+  );
   const { assetId, asset } = props;
   const [assetType, setAssetType] = React.useState("");
-  const [middlewareHostname, setMiddlewareHostname] = React.useState("");
   const [isLoading, setIsLoading] = React.useState(true);
-  const [localipAddress, setLocalIPAddress] = React.useState("");
-  const [ipadrdress_error, setIpAddress_error] = React.useState("");
   const dispatch = useDispatch();
   useEffect(() => {
     setAssetType(asset?.asset_class);
-    setMiddlewareHostname(asset?.meta?.middleware_hostname);
-    setLocalIPAddress(asset?.meta?.local_ip_address);
+    initializeForm(
+      asset?.meta?.middleware_hostname,
+      asset?.meta?.local_ip_address
+    );
     setIsLoading(false);
   }, [asset]);
 
+  const initializeForm = (middleware_hostname: any, local_ip_address: any) => {
+    let form = { ...state.form };
+    form["middleware_hostname"].value = middleware_hostname;
+    form["localipAddress"].value = local_ip_address;
+    dispatch_form({ type: "set_form", form });
+  };
+
+  const initializeError = () => {
+    let form = { ...state.form };
+    form["middleware_hostname"].error = "";
+    form["localipAddress"].error = "";
+    dispatch_form({ type: "set_form", form });
+  };
+
+  const setFieldError = (field: any, error_msg: any) => {
+    let form = { ...state.form };
+    form[field].error = error_msg;
+    dispatch_form({ type: "set_form", form });
+  };
+
+  const isFormValid = () => {
+    initializeError();
+    let form = { ...state.form };
+    if (
+      form["middleware_hostname"].value.trim() !== "" &&
+      form["localipAddress"].value.trim() !== ""
+    ) {
+      if (checkIfValidIP(form["localipAddress"].value)) {
+        return true;
+      } else {
+        setFieldError("localipAddress", "Please Enter a Valid IP address !!");
+        return false;
+      }
+    } else {
+      if (form["middleware_hostname"].value.trim() === "") {
+        setFieldError("middleware_hostname", "This field cannot be empty!!");
+      }
+      if (form["localipAddress"].value.trim() === "") {
+        setFieldError("localipAddress", "This field cannot be empty!!");
+      }
+      return false;
+    }
```

Comment:
Refer implementation from [this](https://github.com/coronasafe/care_fe/blob/develop/src/Components/Patient/PatientRegister.tsx#L496)